pkg/security/ebpf/c/defs.h

```diff
@@ -3,16 +3,136 @@
 
 #include "../../../ebpf/c/bpf_helpers.h"
 
+#if defined(__x86_64__)
+  #define SYSCALL64_PREFIX "__x64_"
+  #define SYSCALL32_PREFIX "__ia32_"
+
+  #define SYSCALL64_PT_REGS_PARM1(x) ((x)->di)
+  #define SYSCALL64_PT_REGS_PARM2(x) ((x)->si)
+  #define SYSCALL64_PT_REGS_PARM3(x) ((x)->dx)
+  #define SYSCALL64_PT_REGS_PARM4(x) ((x)->r10)
+  #define SYSCALL64_PT_REGS_PARM5(x) ((x)->r8)
+  #define SYSCALL64_PT_REGS_PARM6(x) ((x)->r9)
```

Comment:
AFAIK syscalls can only contain up to 5 arguments.

Reply:
It seems many syscalls use 6 arguments (`epoll_pwait`, `io_uring_enter`, `preadv2`, `pwritev2`, `copy_file_range`, ....)

Reply:
It seems `copy_file_range` does use 6. The others use 5. Strange that the https://github.com/torvalds/linux/blob/master/tools/lib/bpf/bpf_tracing.h file in the kernel doesn't have a macro for the 6th argument.

Reply:
Yep. These macros seem to be really for `regular` kprobes and not for `syscall` kprobes